lib/core/test/unit/Cardano/Wallet/CoinSelection/FeeSpec.hs

```diff
@@ -552,98 +510,31 @@ instance Arbitrary (Hash "Tx") where
 
 instance Arbitrary Address where
     shrink _ = []
-    arbitrary = genAddress (30, 100)
-
--- | Generate change addresses for the given network. We consider that change
--- addresses are always following a sequential scheme.
-instance {-# OVERLAPS #-} Arbitrary (Network -> Address) where
-    shrink _ = []
-    arbitrary = do
-        mainnetA <- genAddress (33, 33)
-        testnetA <- genAddress (40, 40)
-        return $ \case
-            Mainnet -> mainnetA
-            Staging -> mainnetA
-            Testnet -> testnetA
-            Local -> testnetA
-
--- | Generate a valid address with a payload of the given size. As pointers,
--- the sizes of payloads are as follows:
---
---     | Network | Scheme     | Size (bytes)   |
---     | ---     | ---        | ---            |
---     | Mainnet | Random     | 72,73,74 or 76 |
---     | Mainnet | Sequential | 39,40,41 or 43 |
---     | Testnet | Random     | 79,80,81 or 83 |
---     | Testnet | Sequential | 46,47,48 or 50 |
---
--- The address format on 'Staging' is the same as 'Mainnet'.
--- The address format on 'Local' is the same as 'Testnet'.
-genAddress :: (Int, Int) -> Gen Address
-genAddress range = do
-    n <- choose range
-    let prefix = BS.pack
-            [ 130       -- Array(2)
-            , 216, 24   -- Tag 24
-            , 88, fromIntegral n -- Bytes(n), n > 23 && n < 256
-            ]
-    payload <- BS.pack <$> vectorOf n arbitrary
-    let crc = CBOR.toByteString (CBOR.encodeWord32 $ crc32 payload)
-    return $ Address (prefix <> payload <> crc)
+    arbitrary = elements
+        [ Address "addr-0"
+        , Address "addr-1"
+        , Address "addr-2"
+        , Address "addr-3"
+        ]
 
 instance Arbitrary CoinSelection where
     shrink sel@(CoinSelection inps outs chgs) = case (inps, outs, chgs) of
         ([_], [_], []) ->
             []
         _ ->
             let
-                inps' = take (max 1 (length inps `div` 2)) inps
-                outs' = take (max 1 (length outs `div` 2)) outs
-                chgs' = take (length chgs `div` 2) chgs
-                inps'' = if length inps > 1 then drop 1 inps else inps
-                outs'' = if length outs > 1 then drop 1 outs else outs
-                chgs'' = drop 1 chgs
+                inps' = if length inps > 1 then drop 1 inps else inps
+                outs' = if length outs > 1 then drop 1 outs else outs
+                chgs' = drop 1 chgs
             in
                 filter (\s -> s /= sel && isValidSelection s)
                     [ CoinSelection inps' outs' chgs'
```

Comment:
I see that we have reduced generator scope of CoinSelection - hope coverage is still big enough to test relevant bits

Reply:
I've only adjusted the shrinker here to be less aggressive and therefore a bit more readable. The more aggressive version is especially needed for the property that has been relocated.

Reply:
yes I see it here https://github.com/input-output-hk/cardano-wallet/pull/212/commits/088c8ccc9e914876ddaca600ac75b462f149761e#diff-80e66ce4975ed1ec822ac0aa910b4553R775